package/MDAnalysis/lib/distances.py

```diff
@@ -1432,26 +1485,39 @@ def calc_bonds(coords1: Union[np.ndarray, 'AtomGroup'],
     .. versionchanged:: 2.3.0
        Can now accept an :class:`~MDAnalysis.core.groups.AtomGroup` as an
        argument in any position and checks inputs using type hinting.
+    .. versionchanged:: 2.5.0
+       Can now optionally use the fast distance functions from distopia
     """
     numatom = coords1.shape[0]
     bondlengths = _check_result_array(result, (numatom,))
 
     if numatom > 0:
         if box is not None:
             boxtype, box = check_box(box)
-            if boxtype == 'ortho':
-                _run("calc_bond_distance_ortho",
-                     args=(coords1, coords2, box, bondlengths),
-                     backend=backend)
+            if boxtype == "ortho":
+                if backend == 'distopia':
```

Comment:
I know this is still messy @IAlibay @richardjgowers and "should" be pushable to the stub. I'm probably missing something super obvious but IMO there is a reason it couldn't be in there. 
- All the **other** distance functions use a pass by reference idiom without assignment and this is reflected in the `run` call as they don't return anything.
-  any use of `.astype()` on the `results` arg forces a copy in the stub function and then we need to return the result. 
- Then the pattern used in `run()` becomes non-consistent with some functions return values while others modify them inplace. making life hell.

What am I missing? As I see it there isn't a way to push the casting into the stub function??? Am I stoopid?

Reply:
Possibly ping @lilyminium @orbeckst if anyone has any ideas. It could just be too late at night.

Reply:
Was this addressed in the end?